app/models/user.rb

```diff
@@ -5,4 +5,6 @@ class User < ApplicationRecord
   devise :database_authenticatable, :rememberable, :trackable, :validatable
 
   devise :omniauthable, omniauth_providers: [:bnet]
+
+  scope :anonymous, -> { find_by_email "anonymous@ghost.com" }
```

Comment:
I think under the hood a scope is a dynamically defined class method, but I don't have any preference. The previous `anonymous` scope was chain-able and returned an active record relation. I think the reason you saw that error is because you used `find_by_email` directly which doesn't return a relation, rather than the scope. Either way, the end result is all the same :D